### PR TITLE
Feat/history view

### PR DESCRIPTION
**Pull request by dcodes05**

### CODE REVIEW

File: libs/oeth/history/src/components/APYContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import { isAddressEqual } from 'viem';
+
+export function APYContainer() {
+  const { address, isConnected } = useAccount();
+  const { data } = useHistoryTableQuery(
+    { addressId: address?.toLowerCase(), offset: 0 },
+    {
+      enabled:
+        isConnected &&
+        isAddressEqual(address, address?.toLowerCase() as `0x${string}`),
```
Comment (toniocodo) [inline libs/oeth/history/src/components/APYContainer.tsx:17]: `isAddressEqual` should be case insensitive, you can omit the lowerCase. Also, there's a helper `HexAddress` type for `0x{string}`

File: libs/oeth/history/src/components/ChartCard.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import {
+  Chart as ChartJS,
+  CategoryScale,
+  LinearScale,
+  LineElement,
+  PointElement,
+} from 'chart.js';
+import { Line } from 'react-chartjs-2';
+
+ChartJS.register(CategoryScale, LinearScale, LineElement, PointElement);
```
Comment (toniocodo) [inline libs/oeth/history/src/components/ChartCard.tsx:14]: Maybe move this to `libs/oeth/shared/src/clients/chart.ts` as it should be called once

File: libs/oeth/history/src/components/Filters.tsx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+                  fill: (theme) => alpha(theme.palette.secondary.main, 0.4),
+                  strokeWidth: '1px',
+                  stroke: (theme) => theme.palette.primary.main,
+                },
+                '&:hover:has(input:not(:checked)) svg': {
+                  fill: (theme) => alpha(theme.palette.secondary.main, 0.4),
+                },
+              }}
+              onChange={(e) =>
+                setSelectedTypes((prev) => {
```
Comment (toniocodo) [inline libs/oeth/history/src/components/Filters.tsx:133]: nit: extract for readability

File: libs/oeth/history/src/components/HistoryButton.tsx [hunk 4]
```diff
@@ -19,49 +19,29 @@ export function HistoryFilterButton({
     <Button
       variant="contained"
       onClick={onClick}
+      disableRipple
```
Comment (toniocodo) [inline libs/oeth/history/src/components/HistoryButton.tsx:22]: should we put this in the theme? I don't know if we want the ripple anywhere...

File: libs/oeth/history/src/components/HistoryCard.tsx [hunk 5]
```diff
@@ -1,83 +1,92 @@
...
-import type { ColumnFilter } from '@tanstack/react-table';
+const PAGE_SIZE = 20;
 
 export function HistoryCard() {
-  const [isConnected, setConnectionState] = useState(false);
+  const [page, setPage] = useState(0);
+  const [filters, setFilters] = useState<string[]>([]);
+  const { address, isConnected } = useAccount();
+
+  const { data, isFetching } = useQuery(
```
Comment (toniocodo) [inline libs/oeth/history/src/components/HistoryCard.tsx:26]: let's use `.graphql` files and codegen for generating gql custom hooks

File: libs/oeth/history/src/components/HistoryCard.tsx [hunk 6]
```diff
@@ -1,83 +1,92 @@
...
+        />
       ) : (
         <Box sx={{ height: '15rem', display: 'grid', placeContent: 'center' }}>
           <Typography>
             {intl.formatMessage({
               defaultMessage: 'Connect your wallet to see your history',
             })}
           </Typography>
-          <Button onClick={() => setConnectionState(true)}>Connect</Button>
+          <Button onClick={() => console.log('test')}>Connect</Button>
```
Comment (toniocodo) [inline libs/oeth/history/src/components/HistoryCard.tsx:87]: 😅

File: libs/oeth/history/src/components/HistoryCell.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+interface Props {
+  timestamp: string;
+  type: 'Sent' | 'Received' | 'Yield' | 'Swap';
+  transactionHash: string;
+}
+
+export function HistoryCell(props: Props) {
+  const intl = useIntl();
+  // @ts-expect-error whatever
```
Comment (toniocodo) [inline libs/oeth/history/src/components/HistoryCell.tsx:23]: 🤣

File: libs/oeth/history/tsconfig.lib.json [hunk 8]
```diff
@@ -20,5 +20,11 @@
     "src/**/*.spec.jsx",
     "src/**/*.test.jsx"
   ],
-  "include": ["src/**/*.js", "src/**/*.jsx", "src/**/*.ts", "src/**/*.tsx"]
+  "include": [
+    "src/**/*.js",
+    "src/**/*.jsx",
+    "src/**/*.ts",
+    "src/**/*.tsx",
+    "src/components/Filters.tsx"
```
Comment (toniocodo) [inline libs/oeth/history/tsconfig.lib.json:28]: 🤔

File: libs/shared/components/src/LinkIcon/index.tsx [hunk 9]
```diff
@@ -1,13 +1,16 @@
...
+import type { SxProps } from '@mui/material';
+
 interface Props {
   url: string;
   size?: string;
+  sx?: SxProps;
 }
 
-export function LinkIcon({ url, size = '0.875rem' }: Props) {
+export function LinkIcon({ url, size = '0.875rem', sx }: Props) {
```
Comment (toniocodo) [inline libs/shared/components/src/LinkIcon/index.tsx:11]: I usually extend full props: 
```
interface Props extends LinKProps {
  ...
}

function LinkIcon({ url, size = '0.875rem', ...rest }) { 
  return <Link {...rest}>
}
```

File: libs/shared/theme/src/theme.tsx [hunk 10]
```diff
@@ -268,10 +278,15 @@ export const theme = extendTheme({
     },
     MuiTableCell: {
       styleOverrides: {
+        // @ts-expect-errors stop acting like it wouldn't work TS
```
Comment (toniocodo) [inline libs/shared/theme/src/theme.tsx:281]: 🤣

File: libs/oeth/history/src/components/HistoryCard.tsx (hunk 5 above)
Comment (dcodes05) [reply libs/oeth/history/src/components/HistoryCard.tsx:26]: I do use it, but I have two different queries and i need to switch between them depending if I have filters or not. so actually the queries are generated using codegen but i need to switch the query dynamically.

File: libs/shared/components/src/LinkIcon/index.tsx (hunk 9 above)
Comment (dcodes05) [reply libs/shared/components/src/LinkIcon/index.tsx:11]: done ✅  changed

File: libs/oeth/history/tsconfig.lib.json (hunk 8 above)
Comment (dcodes05) [reply libs/oeth/history/tsconfig.lib.json:28]: i removed it 👍

File: libs/oeth/history/src/components/APYContainer.tsx (hunk 1 above)
Comment (dcodes05) [reply libs/oeth/history/src/components/APYContainer.tsx:17]: i removed completely this assertion

File: libs/oeth/history/src/components/ChartCard.tsx (hunk 2 above)
Comment (dcodes05) [reply libs/oeth/history/src/components/ChartCard.tsx:14]: done, moved to `shared/providers/chart`

File: libs/oeth/history/src/components/Filters.tsx (hunk 3 above)
Comment (dcodes05) [reply libs/oeth/history/src/components/Filters.tsx:133]: done ✅

**Pull request closed without merging**